src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/prisonperson/PrisonPersonReconService.kt

```diff
@@ -9,28 +9,32 @@ import uk.gov.justice.digital.hmpps.prisonertonomisupdate.config.trackEvent
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomissync.model.PrisonPersonReconciliationResponse
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.prisonperson.model.PhysicalAttributesDto
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.awaitBoth
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.doApiCallWithRetries
 
 @Service
 class PrisonPersonReconService(
   private val dpsApi: PrisonPersonDpsApiService,
-  private val nomisApi: PrisonPersonNomisApiService,
+  private val prisonPersonNomisApi: PrisonPersonNomisApiService,
   private val telemetryClient: TelemetryClient,
 ) {
 
-  suspend fun checkPrisoner(offenderNo: String) = runCatching {
+  suspend fun checkPrisoner(offenderNo: String): String? = runCatching {
     val (nomisPrisoner, dpsPrisoner) = withContext(Dispatchers.Unconfined) {
-      async { nomisApi.getReconciliation(offenderNo) } to
-        async { dpsApi.getPhysicalAttributes(offenderNo) }
+      async { doApiCallWithRetries { prisonPersonNomisApi.getReconciliation(offenderNo) } } to
+        async { doApiCallWithRetries { dpsApi.getPhysicalAttributes(offenderNo) } }
```

Comment:
Add in retries

Reply:
You might want to consider the newer retry mechanism used by e.g. non-associations, it has an 'increasing delay' backoff strategy and more retries, see https://github.com/ministryofjustice/hmpps-prisoner-to-nomis-update/blob/d04d7922fa5b1fe5658baa8f8b9fe23e696cd399/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsApiService.kt#L36